dashboard/public/pixelation/pixelation.js

```diff
@@ -6,13 +6,13 @@
 
 var MAX_SIZE = 400;
 
-var pixel_format, pixel_data, canvas, ctx, widthText, widthRange, heightText, heightRange, bitsPerPixelText, bitsPerPixelRange;
+var pixel_format, pixel_data, canvas, main_ctx, widthText, widthRange, heightText, heightRange, bitsPerPixelText, bitsPerPixelRange, image_w, image_h;
```

Comment:
Nit: At this point, this is a long enough list of global state variables that it would be nice to have some of them broken out with comments explaining how they're used.